### PR TITLE
Accept input from stdin

### PR DESCRIPTION
**Pull request by azizmb**
Just a few minor modifications to allow code input from stdin. 
This makes it easy to use with text editors.

### CODE REVIEW

**Comment by azizmb** [comment]
@asottile I will be updating the test cases, but just wanted some feedback if you're ok with this change.

**Comment by asottile** [comment]
I think this can be done without introducing a special value by using /dev/stdin as a filename.  Then we can just add an extra output format that simply prints.  (It'd have to rewrite a single filename as well)

How does that sound?

**Comment by azizmb** [comment]
@asottile as long as it prints the output don't mind. Though, according to the `argparse` [documentation](https://docs.python.org/3/library/argparse.html#nargs) this seems to be the standard way of doing this.

**Comment by azizmb** [comment]
Also, seems like I forgot to remove some redundant code. The `inline` option I'd used when testing is not actually required. Cleaned that up.

File: reorder_python_imports/main.py [hunk 1]
```diff
@@ -335,8 +337,14 @@ def main(argv=None):
...
-    for filename in args.filenames:
-        contents = io.open(filename).read()
+
+    if not isinstance(args.filenames, list):
+        files = [args.filenames]
+    else:
+        files = args.filenames
+
+    for input_file in files:
+        contents = six.u(input_file.read())
```
Comment (asottile) [inline reorder_python_imports/main.py]: I think this'll break on non-ascii files in python2 (I should add one as a test?)

the lhs handles this using the `io` module

**Comment by asottile** [comment]
Yeah, I'd really rather just add a `--stdout` option which prints to stdout -- then you'd only have to change ~3 lines of code instead.

**Comment by azizmb** [comment]
@asottile I'm not sure I understand. I can add the `--stdout` option, but then I still need to read from `stdin`. How would you prefer I do that?

**Comment by asottile** [comment]
@azizmb `reorder-python-imports /dev/stdin --stdout` I think

**Comment by azizmb** [comment]
Ah ok, updated. Did a force push to cleanup the history, since a lot of the original changes were not needed any more.

Will add test cases if you're ok with this.

The only other thing I was wondering is if it would be better to add some conditional formatting if there is more than one file, like the output of tail. Eg,
`cat somefile.py | reorder-python-imports /dev/stdin --stdout`
output:
```
import operator 
import sys

print 'Hello'
```

`reorder-python-imports somefile.py someotherfile.py --stdout`
output:
```
==> somefile.py <==
import operator 
import sys

print 'Hello'

==> someotherfile.py <==
import abc 
import sys

print 'Hello 2'
```

**Comment by asottile** [comment]
I don't think it's meaningful to put multiple files on stdout.  As precedent, autopep8 only allows multiple files if --in-place or --diff is specified.  Plus you can probably get that to work as-is with xargs:

`ls *.py | xargs --replace bash -c 'echo "==> {} <===" && reorder-python-imports --stdout {}'`

**Comment by asottile** [comment]
Via #15

**Pull request closed without merging**